src/ioc.test.ts

```diff
@@ -81,5 +81,125 @@ describe("ioc tests", () => {
     sinon.assert.calledOnceWithExactly(overrideDependency, "in caller");
     sinon.assert.calledOnceWithExactly(defaultDependency, "in caller");
   });
+  it("setDefault full functionality", async () => {
+    // Setup Dependencies
+    const defaultDependency = sinon.stub();
+    const defaultDependencyObject = {mainDep: defaultDependency};
+    const tooManyEntryDependencyObject = {mainDep: defaultDependency, secondDep: defaultDependency};
+    const additionalDependency = sinon.stub();
+    const additionalDependencyObject = {mainDep: additionalDependency};
+
+    // Setup IOC
+    const iocT5 = iocFactory();
+
+    // setDefault() requires only one default to be set at a time
+    assert.throws(() => {iocT5.setDefault(tooManyEntryDependencyObject);}, {name: 'Error', message: `ioc.setDefault({name}) takes only one thing at a time.`});
+
+    // Successful setDefault() returns iocGet wrapper for the given name & get() basic functionality behaves as expected
```

Comment:
`returns iocGet wrapper for the given name`

Looks like it's actually the result of calling the wrapper, which is just the dependency itself.